cli/src/cli.rs

```diff
@@ -427,9 +432,30 @@ impl CliConfig {
         keypair_path.to_str().unwrap().to_string()
     }
 
+    pub fn default_derivation_path() -> String {
+        "0".to_string()
+    }
+
     pub fn default_json_rpc_url() -> String {
         "http://127.0.0.1:8899".to_string()
     }
+
+    pub(crate) fn pubkey(&self) -> Result<Pubkey, Box<dyn std::error::Error>> {
+        if let Some(path) = &self.keypair_path {
+            if path.starts_with("usb://") {
```

Comment:
This expression is repeated a bunch of times and written as those USB is the one special case. If other protocols are added, the filepath will be the syntactic special case.  Better would be to:

```rust
match parse_keypair_path()?  {
    KeypairUrl::Filepath(keypair_path) => ...,
    KeypairUrl::Usb(stuff_after_usb_colon_slash_slash) => ...,
}
```

Reply:
This is a great suggestion. However, I think it is closely related to the CLI refactor that needs to happen to handle various types of signers (present and not), so I'd like to handle it as part of that work.

Reply:
sounds good